src/rules/trailingCommaRule.ts

```diff
@@ -20,11 +20,98 @@ import * as ts from "typescript";
 
 import * as Lint from "../index";
 
+type OptionValue = "always" | "never" | "ignore";
+
+interface CustomOptionValue {
+    arrays?: OptionValue;
+    exports?: OptionValue;
+    functions?: OptionValue;
+    imports?: OptionValue;
+    objects?: OptionValue;
+    typeLiterals?: OptionValue;
+}
+
 interface Options {
-    multiline?: "always" | "never";
-    singleline?: "always" | "never";
+    multiline?: OptionValue | CustomOptionValue;
+    singleline?: OptionValue | CustomOptionValue;
 }
 
+const defaultOptions: CustomOptionValue = {
+    arrays: "ignore",
+    exports: "ignore",
+    functions: "ignore",
+    imports: "ignore",
+    objects: "ignore",
+    typeLiterals: "ignore",
+};
+
+const normalizeOptions = (options: Options): Options => {
+    if (typeof options.multiline === "string") {
+        options.multiline = {
+            arrays: options.multiline,
+            exports: options.multiline,
+            functions: options.multiline,
+            imports: options.multiline,
+            objects: options.multiline,
+            typeLiterals: options.multiline,
+        };
+    } else {
+        options.multiline = Object.assign({}, defaultOptions, options.multiline);
```

Comment:
Also need to handle the case where the user doesn't specify a `multiline` or `singleline` in the config, in which case we'd just default to an empty object. I'm wondering if explicitly having the `defaultOptions` adds at least a bit of clarity to the code.